### PR TITLE
[5.6.0] Common Error Codes

### PR DESCRIPTION
**Pull request by ngoryachev**
TODO:
- [x] CHANGELOG
- [x] README

### CODE REVIEW

File: android/src/main/java/com/gettipsi/stripe/dialog/AddCardDialogFragment.java [hunk 1]
```diff
@@ -26,11 +26,12 @@
 import com.gettipsi.stripe.StripeModule;
 import com.gettipsi.stripe.util.CardFlipAnimator;
 import com.gettipsi.stripe.util.Utils;
-import com.stripe.android.Stripe;
 import com.stripe.android.TokenCallback;
 import com.stripe.android.model.Card;
 import com.stripe.android.model.Token;
 
+import static com.gettipsi.stripe.Errors.CANCELLED;
+
```
Comment (isnifer) [inline android/src/main/java/com/gettipsi/stripe/dialog/AddCardDialogFragment.java]: Extra whitespace

File: src/Stripe.js [hunk 2]
```diff
@@ -180,6 +168,11 @@ class Stripe {
     )
     return StripeModule.createSourceWithParams(params)
   }
+
+  // eslint-disable-next-line class-methods-use-this
+  get errorCodes() {
+    return errorCodes
+  }
```
Comment (isnifer) [inline src/Stripe.js]: just do it as `static`:
```js
static get errorCodes() {
  return errorCodes
}
```
And pass them to init like this:
```js
StripeModule.init(options, Stripe.errorCodes)
```

File: src/errorCodes.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    errorCode: 'noPaymentRequest',
+    description: 'Missing payment request',
+  },
+  noMerchantIdentifier: {
+    errorCode: 'noMerchantIdentifier',
+    description: ' Missing merchant identifier',
+  },
+  noAmount: {
+    errorCode: 'noAmount',
+    description: 'Amount should be greater 0',
```
Comment (isnifer) [inline src/errorCodes.js]: @densa put all error codes with a description before comment:
```js
// description provided by stripe api
```

File: src/errorCodes.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import { Platform } from 'react-native'
```
Comment (isnifer) [inline src/errorCodes.js]: we do not use `Platform` here as I see

Comment (densa) [reply src/errorCodes.js]: https://github.com/tipsi/tipsi-stripe/blob/533d4773896868f871227d1af1c50e38ca9beee2/src/errorCodes.js#L36

Comment (isnifer) [reply src/errorCodes.js]: OOK

File: src/errorCodes.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  },
+  sourceStatusUnknown: {
+    errorCode: 'sourceStatusUnknown',
+    description: 'Source polling unknown error',
+  },
+  deviceNotSupportsNativePay: {
+    errorCode: 'deviceSupportsApplePay',
+    description: Platform.select({
+      ios: 'This device does not support Apple Pay',
+      android: 'This device does not support Android Pay',
```
Comment (isnifer) [inline src/errorCodes.js]: `Android` => `Google`

File: src/errorCodes.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  sourceStatusFailed: {
+    errorCode: 'sourceStatusFailed',
+    description: 'The source status is unknown. You shouldn\'t encounter this value.',
+  },
+  sourceStatusUnknown: {
+    errorCode: 'sourceStatusUnknown',
+    description: 'Source polling unknown error',
+  },
+  deviceNotSupportsNativePay: {
+    errorCode: 'deviceSupportsApplePay',
```
Comment (isnifer) [inline src/errorCodes.js]: `deviceSupportsApplePay ` => `deviceNotSupportsNativePay`

**Pull request merged**